### PR TITLE
Version handling improvements

### PR DESCRIPTION
**Pull request by gregmuellegger**
Here are some improvements that I made while investigating #1052.
The breakdown is:

* Remove magic constants like `'stable'` and `'latest'` and use a defined constant instead.
* Add `create_stable` / `create_latest` manager methods for easier creation of versions.
* `BetterVersion` is a terrible name IMO :) It should more say about what it actually is. I renamed it to `VersionIdentifier`.
* Added tests that should provide more insight for #1052. Unfortunatelly they didn't. The tests might still be useful to have in the suite.

### CODE REVIEW

**Comment by d0ugal** [comment]
Nice!

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -280,11 +282,11 @@ def save(self, *args, **kwargs):
...
             branch = self.default_branch or self.vcs_repo().fallback_branch
-            if not self.versions.filter(slug='latest').exists():
+            if not self.versions.filter(slug=LATEST).exists():
                 self.versions.create(
-                    slug='latest', verbose_name='latest', machine=True, type='branch', active=True, identifier=branch)
-            # if not self.versions.filter(slug='stable').exists():
-            #     self.versions.create(slug='stable', verbose_name='stable', type='branch', active=True, identifier=branch)
+                    slug=LATEST, verbose_name=LATEST_VERBOSE_NAME, machine=True, type='branch', active=True, identifier=branch)
+            # if not self.versions.filter(slug=STABLE).exists():
+            #     self.versions.create_stable(type='branch', identifier=branch)
```
Comment (ericholscher) [inline readthedocs/projects/models.py]: Looks like the create_stable isn't used currently?

Comment (gregmuellegger) [inline readthedocs/projects/models.py]: It's not in that case. But here: https://github.com/rtfd/readthedocs.org/pull/1354/files#diff-c31d0a7752163caf020c33fb85ba7a96R124

**Comment by ericholscher** [comment]
:fireworks:

**Pull request merged**